cmd/move2kubeapi/move2kubeapi.go

```diff
@@ -17,45 +17,111 @@ limitations under the License.
+	rootCmd.Flags().Bool("auth_enabled", false, "Enable authentication and authorization.")
+	rootCmd.Flags().Bool("secure_cookies", false, "Send cookies only if it is a https TLS connection. Turn this on in production environments.")
+	rootCmd.Flags().String("data_dir", "data", "Path to the directory where all the data will stored. It will be created if it doesn't exist.")
+	rootCmd.Flags().String("static_files_dir", "", "Path to the directory containing static files to be served. Used to serve the Move2Kube UI.")
+	rootCmd.Flags().String("session_secret", "", "A random secret to use for signing session cookies. By default it generates a new session secret.")
+	rootCmd.Flags().String("current_host", "http://localhost:8080", "URL where this server is deployed.")
+	rootCmd.Flags().String("auth_server", "http://localhost:8081", "URL of the authorization server.")
+	rootCmd.Flags().String("auth_server_base_path", "/auth-server", "If the authorization server is hosted under a sub path, specify it here.")
+	rootCmd.Flags().String("auth_server_realm", "testrealm1", "The realm configured in the authorization server.")
+	rootCmd.Flags().String("oidc_discovery_endpoint_path", "", "The OIDC discovery endpoint path for the authorization server. If not specified it uses the default for Keycloak.")
+	rootCmd.Flags().String("uma_configuration_endpoint_path", "", "The UMA configuration endpoint path for the authorization server. If not specified it uses the default for Keycloak.")
+	rootCmd.Flags().String("m2k_client_client_id", "m2k-client", "The OAuth 2.0 client id for the client side.")
+	rootCmd.Flags().String("m2k_client_client_secret", "af10bd64-03e6-47cc-8733-4d04354cf625", "The OAuth 2.0 client secret for the client side.")
+	rootCmd.Flags().String("m2k_client_id_not_client_id", "fb0411ca-3637-4925-9325-9f979bb0e826", "The Id of the client on the Keycloak server. This is NOT the client Id used by OAuth 2.0.")
+	rootCmd.Flags().String("m2k_server_client_id", "m2k-server", "The OAuth 2.0 client id for the server side.")
+	rootCmd.Flags().String("m2k_server_client_secret", "8a1340ff-de5d-42a0-8b40-b6239c7cfc58", "The OAuth 2.0 client secret for the server side.")
+	rootCmd.Flags().String("default_resource_id", "b4d9b0fd-ffdb-4533-9536-5c315af07352", "Resource id on the Keycloak server.")
+	// CloudEvents
+	rootCmd.Flags().Bool("cloud_events_enabled", false, "Enable CloudEvents reporting.")
+	rootCmd.Flags().String("cloud_events_endpoint", "", "Endpoint where CloudEvents are reported.")
+	rootCmd.Flags().String("cloud_events_access_token", "", "Access token to use when reporting CloudEvents.")
+	rootCmd.Flags().String("cloud_events_spec_version", "1.0", "Version of the CloudEvents spec.")
+	rootCmd.Flags().String("cloud_events_type", "", "Type of the CloudEvents event.")
+	rootCmd.Flags().String("cloud_events_subject", "move2kube-api", "Subject to use when reporting CloudEvents.")
```

Comment:
Seems like hyphen is the recommended practise (https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html). We have to check how viper reacts to it though (https://stackoverflow.com/questions/61920435/map-flags-with-words-separated-by-dashes-to-camel-case-config)

Reply:
done